sqlvm-provisioning-csp/SQLServerVMDeployment.json

```diff
@@ -0,0 +1,238 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
```

Comment:
Normally we ask not to provide a default value for parameters that need to be globally unique such as storage account and dns name. Also providing a default value has a side-effect when you try to deploy a template through PS or CLI. When trying to deploy templates using the URI method like in this example below, the client prompts the user for entering a value where a default value is not specified. So if you have an empty string as a default value, the client tools will not prompt and the template deployment will fail 

New-AzureResourceGroupDeployment -Name <deployment-name> -ResourceGroupName <resource-group-name> -TemplateUri https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/arm-asm-s2s/azuredeploy.json
from here http://azure.microsoft.com/en-us/documentation/templates/arm-asm-s2s/